week-3/Homework/mandatory/5-journey-planner.js

```diff
@@ -120,6 +135,24 @@ function getLocationName() { }
   Advanced challange: try to use arrow function when invoking an array method.
 */
 function journeyPlanner(locations, transportMode) {
+  let isAccessiblePlaces=[];
+  let placesIcanGo=[];
+  let transport = locations . map (getTransportModes);
+  
+  for( var i = 0; i < transport.length;i++ ){
+   
+    if (isAccessibleByTransportMode(transport[i],transportMode)) { 
+      isAccessiblePlaces.push(locations[i]);
+     
+    }
+  }
```

Comment:
Another way to achieve this is to use array methods.
 1) Use `filter` on the `locations` list then inside the filter function you can 
     - get the transport mode of that location: `let transports = getTransportModes(location)
     - then can call isAccessibleByTransportMode(transports,transportMode) with the transport modes above.
 2) Finally, you have to just map over the filtered list of locations, there you can reuse the `getLocationName` function, similarly as you used the `getTransportModes` on map.

Your code would like something like this:
```
locations.filter(function(location) {
 // here your filter logic comes as described in point 1.
}).map(getLocationName);
```